src/sentry/models/group.py

```diff
@@ -45,6 +54,28 @@ class GroupStatus(object):
 class GroupManager(BaseManager):
     use_for_related_fields = True
 
+    def by_qualified_short_id(self, org, short_id):
+        match = _short_id_re.match(short_id.strip())
+        if match is None:
+            return
```

Comment:
This is just my opinion, so feel free to disregard.

For context, I am used to doing code reviews where the reviewer tries to give constructive feedback rather than just looking for bugs. I think that is a process that is worthwhile for both the author and the reviewer and results in a lower bus factor, a cleaner, more consistent codebase, and a more maintainable product. Maybe that's not the culture that other people want to promote here. I would like to. We probably should talk internally about what the expectations are here.

As far as the comment on the actual code, I don't think — again, this is personal opinion — methods on things like manager classes should be responsible for performing data validation. The method signature should be structured in a way that invalid data can't be passed to it. The validation obviously needs to be performed somewhere. I just don't think it is here.